Ruins/src/main/resources/examplesAndDocs/template_rules.txt

```diff
@@ -284,6 +284,117 @@ preserve_plants=0
 # Example: rule1=0,100,teBlock;minecraft:trapped_chest;{x:-156,y:65,z:72,Items:[0:{Slot:12b,id:5s,Count:1b,Damage:0s},1:{Slot:13b,id:24s,Count:1b,Damage:1s}],id:"Chest"}-2
 # Note: "Chain" and "Repeat" Command Blocks are classified as such by default
 #
+# --- begin proposal ---
+#
+# The rule syntax was modified a bit to add some new features (variants and variant groups, in particular). None of these changes
+# should break or alter the behavior of existing template files.
+#
+# Block Weighting: Instead of repeating block IDs in a rule to manipulate their probabilities of occurrence, weighting factors may
+# now be used. This can make rules shorter and, in certain circumstances, more efficient. Apply a weight to an ID in a rule by
+# preceding it with a prefix of the form "<weight>*"; this is functionally equivalent to repeating the ID <weight> times. Valid
+# values are integers between 1 and 99999, inclusive. For example, the following rules are essentially identical, each producing
+# blocks of cobble, mossy cobble, and gravel with roughly 56%, 33%, and 11% likelihood, respectively:
+#     rule1=0,100,cobblestone,cobblestone,mossy_cobblestone,cobblestone,gravel,cobblestone,mossy_cobblestone,mossy_cobblestone,cobblestone
+#     rule2=0,100,5*cobblestone,gravel,3*mossy_cobblestone
+# Weights are optional. If no weight is specified for an ID in a rule, a weight of 1 is assumed.
+#
+# Rule Variants: Consider a platform composed of many instances of the following rule3:
+#     rule3=0,100,wool-14,wool-4,wool-11
```

Comment:
You should write down what effect each meta stands for so someone who doesn't know ruins can figure out the math
I mean, i barely grasp it and i know how the weights work

Reply:
I'm not sure I understand. In this rule, the 14, 4, and 11 are standard Minecraft wool color metadata values for red, yellow, and blue. They've got nothing to do with weights. This rule doesn't use any of the features I added; it (and the accompanying text) is only here to illustrate the difference between per-block random selection (what the current mod provides) and per-structure random selection (the whole motivation for rule variants).

Reply:
Lesser beings don't see "red, yellow, and blue wool blocks". They see "wool-14,wool-4,wool-11" and might not make the connection. I annotated it here, but later when you do math examples it is the same, you mention different wood types which people might not know which blocks and meta values they represent.